client/my-sites/stats/activity-log-day/index.jsx

```diff
@@ -41,15 +42,38 @@ class ActivityLogDay extends Component {
 		// Connected props
 		isToday: PropTypes.bool.isRequired,
 		recordTracksEvent: PropTypes.func.isRequired,
+		requestedRewind: PropTypes.string,
 	};
 
 	static defaultProps = {
 		disableRestore: false,
 		isRewindActive: true,
 	};
 
+	state = {
+		rewindHere: false,
+		dayExpanded: false,
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		// if Rewind dialog is being displayed and it's then canceled or a different Rewind button is clicked
+		if (
+			this.state.rewindHere &&
+			( isEmpty( nextProps.requestedRewind ) ||
+				this.props.requestedRewind !== nextProps.requestedRewind )
```

Comment:
this `isEmpty()` check may be redundant. Are we only considering cases where it _was_ something but now _isn't_? An empty `nextProps.requestedRewind` will already `!==` a filled-in `this.props.requestedRewind`

also, what is the type of `requestedRewind`? A string? an object?

Reply:
> this isEmpty() check may be redundant.

Good catch. I've removed the isEmpty.

> what is the type of requestedRewind?

It's a string. It's added in the prop types.